test/support/factories.rb

```diff
@@ -21,16 +21,16 @@
     sequence(:w_id){ |n| n}
   end
 
-  factory :topic_with_book, :parent=>:topic do |m|
+  factory :topic_with_book, parent::topic do |m|
```

Comment:
I think there needs to be a space above otherwise ruby fails to parse this line.